### DISCUSSION TITLE
Table callback called twice

### DISCUSSION THREAD

**Discussion by dimitrove** [fltk-rs/fltk-rs#817] [Q&A] [answered]
I have button callback and table callback:

    btn.set_callback(move |b| {
            let trigger = b.trigger();
	        println!("button-trigger: {:?}", trigger);
        }
    );

    tbl.set_callback(move |t| {
        let trigger = t.trigger();
        let ctx = t.callback_context();
        let row = t.callback_row();
        let col = t.callback_col();
        println!("table-trigger-ctx-row-col: {:?}-{:?}-{}-{}", trigger, ctx, row, col);
    });

What I get:
- button - left-mouse-button-pressed - NOTHING
- button - left-mouse-button-released - button-trigger: Release

- table - left-mouse-button-pressed -
table-trigger-ctx-row-col: Release-Cell-1-0

- table - left-mouse-button-released -
table-trigger-ctx-row-col: Release-Cell-1-0

How can I get the callback called once only?

**Comment by MoAlyousef** [comment, marked as answer]
FLTK hardcodes the do_callback() in the handle method of the table, for Events Push and Released when the CallbackTrigger is Release.
If you don't want to modify the CallbackTrigger for the table, and are only interested in the Release of the mouse click, you can ignore the Push event in your callback:
```rust
        if app::event() == enums::Event::Push {
            return;
        }
```
